### PR TITLE
feat: redesign InfoCards component and update schema

### PR DESCRIPTION
**Pull request by dcshzj**
## Problem

InfoCards component has drifted from the [Figma designs](https://www.figma.com/design/rsKQdmtyoOWawyAv1LkJJK/(New%2C-WIP)-Isomer-Next-Component-Library?node-id=1-13&t=FBIqtmtr8UmVuVGa-0).

## Solution

**Breaking Changes**

- [x] Yes - this PR contains breaking changes
    - This will require schema changes and migrations!
- [ ] No - this PR is backwards compatible

**Improvements**:

- Redesigned InfoCards component to remove the button label.
- Added no image variant of InfoCards.
- Updated schema to enforce that all cards have or don't have images.
- Also updated JSONForms boolean control to avoid rendering if there is a const present, similar to how we do for `type` properties.
- Added the new InfoCards component to the auto-generated index page.

### CODE REVIEW

File: apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsBooleanControl.tsx
Comment (harishv7) [inline apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsBooleanControl.tsx:29]: what's schema.const?

File: apps/studio/src/components/PageEditor/constants.ts
Comment (harishv7) [inline apps/studio/src/components/PageEditor/constants.ts:87]: do you think there can be other states for cards? if so we can use an enum like "WithImages, TextOnly" etc. and is extensible?

File: apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsBooleanControl.tsx
Comment (dcshzj) [reply apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsBooleanControl.tsx:29]: `schema.const` is a JSON Schema construct where the particular property takes in a specific literal (can be string/number/boolean). In this case, if the `const` is defined (say it's `true) then it is not meaningful to show the toggle, since the `false` case is invalid.

File: apps/studio/src/components/PageEditor/constants.ts
Comment (dcshzj) [reply apps/studio/src/components/PageEditor/constants.ts]: Ah good point, this was the original based on designs where there was a toggle button but this has changed to use a dropdown to select the variants. Adjusted in 34dcbfd2de27c92504ddae29f4845f7835d1a8b0!

File: apps/studio/prisma/generated/generatedEnums.ts
Comment (dcshzj) [reply apps/studio/prisma/generated/generatedEnums.ts:1]: Rebase removed le

File: packages/components/src/templates/next/components/complex/InfoCards/InfoCards.tsx
Comment (karrui) [inline packages/components/src/templates/next/components/complex/InfoCards/InfoCards.tsx:157]: nit: should we strengthen the typechecking so build-time errors will surface if we add more variants?

```tsx
// The typing forces compile time rendering to prevent `undefined` from being returned,
// which can happen if variants change or is added
const InfoCardToRender: ReactElement = useMemo(() => {
  switch (variant) { // I assuming variant is typed like `"cardsWithImages" | "cardsWithoutImages"`
    case "cardsWithImages"
      return InfoCardWithImage
    case "cardsWithoutImage"
      return InfoCardNoImage
  }
}, [variant])

// Usage
{cards.map((card, idx) => (
  <InfoCardToRender ... />
)}
```

File: tooling/build/scripts/generate-sitemap.js
Comment (karrui) [inline tooling/build/scripts/generate-sitemap.js:1]: i forgot to put formatter in this package LOL! please format :P

**Review by karrui — APPROVED** [review]
lgtm. Leave it to @sehyunidaaa to review chromatic

File: packages/components/src/templates/next/components/complex/InfoCards/InfoCards.tsx
Comment (dcshzj) [reply packages/components/src/templates/next/components/complex/InfoCards/InfoCards.tsx]: Updated in 1d3845efe6d4a8db0faf455d1c260a04b0d37ea1, can't really switch case after the map cos the type doesn't narrow properly.

File: tooling/build/scripts/generate-sitemap.js
Comment (dcshzj) [reply tooling/build/scripts/generate-sitemap.js:1]: Updated in f942480e91cc879bb434efe219b2b2548aa6ec40.

**Comment by dcshzj** [comment]
Handing over this PR to @sehyunidaaa

**Comment by sehyunidaaa** [comment]
Working on rest of the styles, please don't merge :)

**Comment by sehyunidaaa** [comment]
Updated with styling tweaks. Please re-review!

File: packages/components/src/templates/next/components/complex/InfoCards/InfoCards.tsx
Comment (sehyunidaaa) [inline packages/components/src/templates/next/components/complex/InfoCards/InfoCards.tsx:82]: @dcshzj I had to pass `icClickableCard` here as well because the hover effect applies to both `InfoCardImage` (hover on image) AND `cardTitle` (color change, arrow moves). Previously `group-hover` wasn't working on the `InfoCardImage`. Hope this is not weird in terms of practice though?

Comment (dcshzj) [reply packages/components/src/templates/next/components/complex/InfoCards/InfoCards.tsx:82]: Yup this works!

File: tooling/build/package.json
Comment (karrui) [inline tooling/build/package.json:14]: nit: might need an extra 

```
  "prettier": "@isomer/prettier-config"
```

so formatting works properly

**Review by karrui — APPROVED** [review]
lgtm if design passes design

File: apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsBooleanControl.tsx
Comment (dcshzj) [reply apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsBooleanControl.tsx]: Removed in 2e49116ea563bf870418089a4162e584403c1e3c.

File: tooling/build/package.json
Comment (dcshzj) [reply tooling/build/package.json]: Added in 2e49116ea563bf870418089a4162e584403c1e3c.

File: tooling/build/package.json
Comment (karrui) [inline tooling/build/package.json:15]: need run formatting again since all tooling affected hehehe

Comment (dcshzj) [reply tooling/build/package.json:15]: Ah okay for some reason it didn't give me anything the first time round, fixed in d2a6419be0ac8e1ae89e1d554c3c351a50f9c8e2.

**Comment by karrui** [comment]
ISOM-1272

**Pull request merged**
